src/main/java/eu/proteus/producer/model/SensorMeasurement.java

```diff
@@ -12,7 +12,7 @@
 	protected double x;
 
 	public SensorMeasurement() {
-		this.type = this.getClass() == SensorMeasurement2D.class ? (byte) 0x0001f : (byte) 0x0000f;
+		this.type = this.getClass() == SensorMeasurement2D.class ? (byte) 0x0001 : (byte) 0x0000;
```

Comment:
Because JVM always signed all types, and it's a good practice to cast always hex strings to byte just before working with that. Casting always keep only the last 8 bits on hex strings